capture.py

```diff
@@ -309,86 +323,86 @@ def _independent_panel(width, height):
         # Delete the panel to fix memory leak (about 5 mb per capture)
         cmds.deleteUI(panel, panel=True)
         cmds.deleteUI(window)
-
-
-@contextlib.contextmanager
-def _applied_viewport_options(options, panel):
-    """Context manager for applying `options` to `panel`"""
-
-    options = dict(ViewportOptions, **(options or {}))
-
-    cmds.modelEditor(panel,
-                     edit=True,
-                     allObjects=False,
-                     grid=False,
-                     manipulators=False)
-    cmds.modelEditor(panel, edit=True, **options)
-
-    yield
-
-
-@contextlib.contextmanager
-def _applied_camera_options(options, panel, camera):
-    """Context manager for applying `options` to `camera`"""
-
-    options = dict(CameraOptions, **(options or {}))
-
-    old_options = dict()
-    for opt in options:
-        try:
-            old_options[opt] = cmds.getAttr(camera + "." + opt)
-        except:
-            sys.stderr.write("Could not get camera attribute "
-                             "for capture: %s" % opt)
-            delattr(options, opt)
-
-    for opt, value in options.iteritems():
-        cmds.setAttr(camera + "." + opt, value)
-
-    try:
-        yield
-    finally:
-        if old_options:
-            for opt, value in old_options.iteritems():
-                cmds.setAttr(camera + "." + opt, value)
-
-
+        
+        
+def parse_active_view():
+    """Parse the current settings from the active view"""
+    
+    panel = cmds.getPanel(wf=True)
+    camera = cmds.modelEditor(panel, q=1, camera=1)
+    return parse_view(panel, camera)
+
+        
+def parse_view(panel, camera):
+    """Parse the scene, panel and camera for their current settings"""
+
+    # Display options
+    display_options = {}
+    for key in DisplayOptions.keys():
```

Comment:
Was actually the only reason why I though keeping the `viewport_options` and `display_options`, etc. separate so that we could use each option type with its own command.